### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by Corginyan** [pixijs/pixi-sound#94]
I trying to stop or destroy sound instance before play end. But it's resulting in probable memory leak. In chrome heap dump i see multiple objects of `AudioBufferSourceNode ` in `Pending activities`. Seems like that objects is can't be properly garbage collected. Only workaround i found it's mute sound and wait until it's ends in background.

```
PIXI.loader.add('testSound', 'stest/sound.ogg');

PIXI.loader.load((loader, res) => {
    let sound = res.testSound.sound;

    let instances = [];

    for (let i = 0; i < 30; i++) {
        let instance = sound.play({loop: true});
        instances.push(instance)
    }

    setTimeout(() => {
        // Nothing helps
        // instances.forEach(instance => {
        //
        //     if (instance._source) {
        //         instance._enabled = false;
        //         instance._source.onended = null;
        //         instance._source.loop = false;
        //         instance._source.loopStart = 0;
        //         instance._source.loopEnd = 0;
        //         instance._source.stop(0);
        //         instance._source.disconnect(0);
        //     }
        //
        //     instance.stop();
        //     instance.destroy()
        // });
        // PIXI.sound.removeAll();
        // PIXI.sound.stopAll();

        // Workaround
        instances.forEach(instance => {
            instance.muted = true;
            instance.loop = false;
        });

        instances = []
    }, 3000)
});
```

**Comment by djmisterjon** [comment]
hi, hum interesting, i confirm memory leak.
It seem some reference prevent the GC .
thank for the tips.
Need now found why some reference are not clear and prevent GC.
![image](https://user-images.githubusercontent.com/24865815/56687159-ad9d9980-66a3-11e9-8ff6-f36d8dfa55ca.png)

**Comment by bigtimebuddy** [comment]
@djmisterjon one good place to start would be @Corginyan's suggestion in #95

**Comment by djmisterjon** [comment]
@bigtimebuddy 
Ok but I thought it was problematic, because in the diag of memory, these the only reference that exists
![image](https://user-images.githubusercontent.com/24865815/56691667-85676800-66ae-11e9-81e7-4bf215d9a04f.png)
But should not it disappear after playing an instance?
I may not understand the use of the pool here. Would not it be similar to the Tweens system in Greensock?
If I'm not mistaken, they have their own GC emptying their pool after ~2sec, which is then dumped by the GC.

**Comment by bigtimebuddy** [comment]
The pool is there by design, however, a purge of the pool every few seconds isn't a terrible idea.

**Comment by djmisterjon** [comment]
ok fine, i was think it was similar logic with tween. my bad.
I know greensock use a kind of pool when use multiple instance, and than after ~x sec, it release animations for GC.﻿
It what i was understood when I asked the question about a memory leak from multi instance.
I thinking there was a bit of similarity here, but I'm wrong. :) oups
https://greensock.com/forums/topic/19470-garbage-collector-and-manage-memory/?tab=comments#comment-90426

**Comment by Corginyan** [comment]
Seems like it's bug in chrome itself. I don't fully understand Web Audio API, but that code produce same behaviour.
```
const context = new AudioContext();

for (let i = 0; i < 1000; i++) {
    let node = context.createBufferSource();
    node.start();
    node.stop();
}
```

**Comment by djmisterjon** [comment]
@Corginyan 
it seem work if you do 
```javascript
context = new AudioContext();

for (let i = 0; i < 1000; i++) {
    let node = context.createBufferSource();
    node.start();
    node.stop();
}
context.close()
```
work for me on nwjs with  chromium.
https://www.chromestatus.com/feature/5411681005993984
https://googlechrome.github.io/samples/webaudio-audiocontext-close/index.html

**Comment by Corginyan** [comment]
So, as i can see, one of probable solution is to create separate `AudioContext` for every `AudioBufferSourceNode`, so we can `close()` it at any time after call `stop()` for `AudioBufferSourceNode` just to wipe that objects from memory. I think it's really hard to implement for current library architecture.

**Comment by djmisterjon** [comment]
yes and also this will allow custom filters for each instance.. 
let see what @bigtimebuddy think about this.
it seems to me indeed complicated to correct here, but esencial to avoid memory leak.
`separate AudioContext for every ` will remove the optimisation he made for the plugin. but i think it will fix memory leak.

**Comment by Corginyan** [comment]
Looks like chrome developers fixed that memory leak in chrome canary 79.0.3933.1

**Comment by Corginyan** [comment]
No problem in latest chrome and firefox.
